### ISSUE TITLE
Remove Bazel warnings about canonical reproducible forms for Cargo crates

### ISSUE THREAD

**Issue by daviddrysdale** [project-oak/oak#851]
#801 added the use of `cargo raze` to generate Bazel `BUILD` files for Rust crates that are needed as dependencies by the Rust runtime.

However, the generated `cargo/crates.bzl` file doesn't include `sha256` values for the packages it is setting up.  This seems to be because the `cargo-raze` tool generates this information from the output of `cargo metadata`, which [doesn't have a package hash](https://doc.rust-lang.org/cargo/commands/cargo-metadata.html).

```log
(06:52:42) INFO: SHA256 (https://crates-io.s3-us-west-1.amazonaws.com/crates/pin-project-lite/pin-project-lite-0.1.4.crate) = 237844750cfbb86f67afe27eee600dfbbcb6188d734139b534cbfbf4f96792ae
(06:52:42) DEBUG: Rule 'raze__pin_project_lite__0_1_4' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "237844750cfbb86f67afe27eee600dfbbcb6188d734139b534cbfbf4f96792ae"
(06:52:42) DEBUG: Call stack for the definition of repository 'raze__pin_project_lite__0_1_4' which is a http_archive (rule definition at /workspace/bazel-cache/clang/external/bazel_tools/tools/build_defs/repo/http.bzl:292:16):
 - /workspace/cargo/crates.bzl:11:9
 - /workspace/cargo/crates.bzl:478:5
 - /workspace/WORKSPACE:266:1
```

**Comment by tiziano88** [comment]
I hope "removing the warnings" means adding the sha256 to the rules in an automatic way, rather than just silencing them? BTW are those hash in sync with those in Cargo.lock?

**Comment by daviddrysdale** [comment]
No longer relevant after #1016.
